crypto/sm2/sm2_pmeth.c

```diff
@@ -63,6 +69,16 @@ static int pkey_sm2_copy(EVP_PKEY_CTX *dst, EVP_PKEY_CTX *src)
             return 0;
         }
     }
+    if (sctx->id != NULL) {
+        dctx->id = OPENSSL_malloc(sctx->id_len);
+        if (dctx->id == NULL) {
+            pkey_sm2_cleanup(dst);
```

Comment:
This looks like it is missing an SM2err(SM2_F_PKEY_SM2_COPY, ERR_R_MALLOC_FAILURE) call ... for consistency with the other error handling.